### PR TITLE
148 upvot downvote functions

### PR DESCRIPTION
**Pull request by yishain11**

### CODE REVIEW

File: frontend/src/pages/AgreementPage.tsx [hunk 1]
```diff
@@ -7,12 +7,16 @@ import SectionCard from 'components/SectionCard';
...
+import { DataContext } from 'contexts';
 
 const AgreementPage: FC = () => {
   const { t } = useTranslation();
+  const { user } = useContext(DataContext);
+  const userId: number | undefined = user?.id;
   const agreementContext = useContext(AgreementContext);
   const currentCategory = agreementContext.categoryName;
   const agreement: IAgreement | undefined = agreementContext.agreement;
+  console.log('agreement', agreement);
```
Comment (avirankatz) [inline frontend/src/pages/AgreementPage.tsx:19]: Could you please remove this line?

**Review by avirankatz — COMMENTED** [review]
Nice PR.
- [ ] remove userId prop. instead get it inside agreement context (use data context inside agreement context)

File: frontend/src/contexts/agreement.tsx [hunk 2]
```diff
@@ -15,13 +16,46 @@ const AgreementProvider: FC = () => {
...
+
   const agreement = data?.core_agreements[0];
   const state: IAgreementContext = {
     agreementId: agreement?.id || NaN,
     categoryName: agreement?.category?.name || '',
     rationale: agreement?.rationale || '',
     agreementTitle: agreement?.name || '',
     agreement: agreement,
+    vote: (user_id: number, version_id: number, type: 'up' | 'down') => {
+      return onVote(user_id, version_id, type);
```
Comment (avirankatz) [inline frontend/src/contexts/agreement.tsx:57]: Please refactor this to be a one-liner

File: frontend/src/contexts/agreement.tsx [hunk 3]
```diff
@@ -15,13 +16,46 @@ const AgreementProvider: FC = () => {
       agreementId,
     },
   });
+
+  const [voteFn, { error: voteError }] = useMutation(voteMutation, {
+    refetchQueries: ['votes'],
+  });
+
+  const onVote = async (user_id: number, version_id: number, type: 'up' | 'down') => {
```
Comment (avirankatz) [inline frontend/src/contexts/agreement.tsx:24]: Let's rename this to just "vote", as this isn't an event handler

File: frontend/src/contexts/agreement.tsx [hunk 4]
```diff
@@ -15,13 +16,46 @@ const AgreementProvider: FC = () => {
       agreementId,
     },
   });
+
+  const [voteFn, { error: voteError }] = useMutation(voteMutation, {
+    refetchQueries: ['votes'],
```
Comment (avirankatz) [inline frontend/src/contexts/agreement.tsx:21]: This is meaningless at this point, because we don't have a query called 'votes' (not yet at least)

**Review by avirankatz — CHANGES_REQUESTED** [review]
Nice one. I'm still loving the short PRs. Also, the vote api is good and comfortable.
- [ ] Remove userId from section and vote function
- [ ] use DataContext inside agreement context and handle user id from there
- [ ] rename onVote to vote
- [x] one-liner function inside agreement context

**Review by avirankatz — APPROVED** [review]

**Review by avirankatz — CHANGES_REQUESTED** [review]
Sorry, approved by mistake (see comments from last requested changes)

File: frontend/src/contexts/agreement.tsx (hunk 2 above)
Comment (yishain11) [reply frontend/src/contexts/agreement.tsx:57]: its too long, so the linter turns it into multiple lines

File: frontend/src/contexts/agreement.tsx (hunk 3 above)
Comment (yishain11) [reply frontend/src/contexts/agreement.tsx:24]: done

File: frontend/src/contexts/agreement.tsx (hunk 4 above)
Comment (yishain11) [reply frontend/src/contexts/agreement.tsx:21]: removed

**Pull request merged**